pkg/autodiscovery/providers/utils.go

```diff
@@ -100,6 +101,25 @@ func buildTemplates(key string, checkNames []string, initConfigs, instances []in
 // extractTemplatesFromMap looks for autodiscovery configurations in a given map
 // (either docker labels or kubernetes annotations) and returns them if found.
 func extractTemplatesFromMap(key string, input map[string]string, prefix string) ([]integration.Config, error) {
+	configs := make([]integration.Config, 0)
+
+	checksConfigs, err := extractCheckTemplatesFromMap(key, input, prefix)
+	if err != nil {
+		return []integration.Config{}, err
+	}
+	configs = append(configs, checksConfigs...)
+
+	logsConfigs, err := extractLogsTemplatesFromMap(key, input, prefix)
+	if err != nil {
+		return []integration.Config{}, err
```

Comment:
I think you should return `configs` there to at least have the check configs if log fails